pandas/core/arrays/string_.py

```diff
@@ -73,51 +80,129 @@ class StringDtype(ExtensionDtype):
     Examples
     --------
     >>> pd.StringDtype()
-    StringDtype
+    string[python]
```

Comment:
mention that the output repr changed in the release note as well

Reply:
with latest changes, no longer applicable

Reply:
now that the name and the str() repr and name has changed, we could maybe change this back to StringDtype and add the stroage.

i.e `StringDtype(storage="python")`